partiql-tests-data/eval/primitives/functions/abs.ion

```diff
@@ -1,256 +1,259 @@
+      output: 0d0
     }
   },
   {
-    name:"ABS(`-0d0`)",
-    statement:"ABS(`-0d0`)",
-    assert:{
-      result:EvaluationSuccess,
-      evalMode:[
+    name: "ABS(`-0d0`)",
+    statement: "ABS(`-0d0`)",
+    assert: {
+      result: EvaluationSuccess,
+      evalMode: [
         EvalModeCoerce,
         EvalModeError
       ],
-      output:0d0
+      output: 0d0
     }
   },
   {
-    name:"ABS(`-0d-0`)",
-    statement:"ABS(`-0d-0`)",
-    assert:{
-      result:EvaluationSuccess,
-      evalMode:[
+    name: "ABS(`-0d-0`)",
+    statement: "ABS(`-0d-0`)",
+    assert: {
+      result: EvaluationSuccess,
+      evalMode: [
         EvalModeCoerce,
         EvalModeError
       ],
-      output:0d0
+      output: 0d0
     }
   },
   {
-    name:"ABS(`-0.0d1`)",
-    statement:"ABS(`-0.0d1`)",
-    assert:{
-      result:EvaluationSuccess,
-      evalMode:[
+    name: "ABS(`-0.0d1`)",
+    statement: "ABS(`-0.0d1`)",
+    assert: {
+      result: EvaluationSuccess,
+      evalMode: [
         EvalModeCoerce,
         EvalModeError
       ],
-      output:0d0
+      output: 0d0
     }
   },
   {
-    name:"ABS(`-0.0000`) preserve scale",
-    statement:"ABS(`-0.0000`)",
-    assert:{
-      result:EvaluationSuccess,
-      evalMode:[
+    name: "ABS(`-0.0000`) preserve scale",
+    statement: "ABS(`-0.0000`)",
+    assert: {
+      result: EvaluationSuccess,
+      evalMode: [
         EvalModeCoerce,
         EvalModeError
       ],
-      output:0d-4
+      output: 0d-4
     }
   },
   {
-    name:"ABS(`-inf`) special value",
-    statement:"ABS(`-inf`)",
-    assert:{
-      result:EvaluationSuccess,
-      evalMode:[
+    name: "ABS(`-inf`) special value",
+    statement: "ABS(`-inf`)",
+    assert: {
+      result: EvaluationSuccess,
+      evalMode: [
         EvalModeCoerce,
         EvalModeError
       ],
-      output:+inf
+      output: +inf
     }
   },
   {
-    name:"ABS(`+inf`) special value",
-    statement:"ABS(`+inf`)",
-    assert:{
-      result:EvaluationSuccess,
-      evalMode:[
+    name: "ABS(`+inf`) special value",
+    statement: "ABS(`+inf`)",
+    assert: {
+      result: EvaluationSuccess,
+      evalMode: [
         EvalModeCoerce,
         EvalModeError
       ],
-      output:+inf
+      output: +inf
     }
   },
   {
-    name:"ABS(`nan`) special value",
-    statement:"ABS(`nan`)",
-    assert:{
-      result:EvaluationSuccess,
-      evalMode:[
+    name: "ABS(`nan`) special value",
+    statement: "ABS(`nan`)",
+    assert: {
+      result: EvaluationSuccess,
+      evalMode: [
         EvalModeCoerce,
         EvalModeError
       ],
-      output:nan
+      output: nan
     }
   },
   {
-    name:"ABS(MISSING) null propogation",
-    statement:"ABS(MISSING)",
-    assert:{
-      result:EvaluationSuccess,
-      evalMode:[
-        EvalModeCoerce,
-        EvalModeError
-      ],
-      output:$missing::null,
-    }
+    name: "ABS(MISSING) null propogation",
+    statement: "ABS(MISSING)",
```

Comment:
Perhaps we can clarify this behavior in the spec or some other document before changing the behavior? As the PartiQL spec is currently written, it's unclear to me if these cases would be an error in strict mode.

For most functions/operations (other than equality, `IS MISSING`, and some boolean ops), a data type mismatch would result in an error in strict mode and return `MISSING` in permissive. E.g.
- `ABS('some string')`
  - `MISSING` in permissive
  - error in strict

I guess it's less clear to me if a `MISSING` literal denotes the same erroring behavior. E.g.
- `ABS(MISSING)`
  - `MISSING` in permissive
  - error or `MISSING` in strict?

Following @jpschorr's analogy of signaling and quiet [NaNs](https://en.wikipedia.org/wiki/NaN), the `MISSING` would have two different types
- Signaling `MISSING` -> data type mismatches, path expression yielding no field
- Quiet `MISSING` -> literal `MISSING` in the query text or input data

If we are following this signaling vs quiet model, then I would think quiet `MISSING`s as a function argument would return a quiet `MISSING` in strict mode.

Reply:
It's interesting that the spec is very explicit for the cases in which strict mode would give an error on `MISSING` like [Section 5.1.1 Mistyping Cases](https://partiql.org/partiql-lang/#sec:bag-array-mistypings) but is less explicit for function arguments in section 7.

The spec does give a direct example using the `MISSING` literal in an operation:
> Example 28. Each one of these expressions returns MISSING: 5 + missing

but it's unclear whether this applies to both permissive and strict modes.